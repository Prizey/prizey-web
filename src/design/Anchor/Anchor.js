import React, { useState } from 'react'
import { Link, navigate } from '@reach/router'
import ConfirmLeaveDialog from '../ConfirmLeaveDialog'

export const handleClose = setIsOpen => () => setIsOpen(false)
export const handleClick = (setIsOpen, confirmLeave) => evt => {
  if (confirmLeave) {
    evt.preventDefault()
    setIsOpen(true)
  }
}
export const handleOnConfirm = to => () => navigate(to)

export default ({ to, children, confirmLeave, ...props }) => {
  const [isOpen, setIsOpen] = useState(false)

  if (to.startsWith('http')) {
    return (
      <React.Fragment>
        {confirmLeave && (
          <ConfirmLeaveDialog
            isOpen={isOpen}
            close={handleClose(setIsOpen)}
            confirm={handleOnConfirm(to)}
          />
        )}
        <a href={to} {...props} onClick={handleClick(setIsOpen, confirmLeave)}>
          {children}
        </a>
      </React.Fragment>
    )
  }

  return (
    <React.Fragment>
      {confirmLeave && (
        <ConfirmLeaveDialog
          isOpen={isOpen}
          close={handleClose(setIsOpen)}
          confirm={handleOnConfirm(to)}
        />
      )}
      <Link to={to} {...props} onClick={handleClick(setIsOpen, confirmLeave)}>
        {children}
      </Link>
    </React.Fragment>
  )
}
