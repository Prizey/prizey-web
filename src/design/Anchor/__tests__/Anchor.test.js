import React from 'react'
import renderer from 'react-test-renderer'
import Anchor from '../Anchor'

it('renders correctly', () => {
  const params = {
    user: { tickets: 10 },
  }
  const tree = renderer.create(<Anchor {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('get the user from croods state', () => {
  // handleClick()()
})
