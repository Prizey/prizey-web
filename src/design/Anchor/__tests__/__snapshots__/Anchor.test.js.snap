// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the component is mounted renders correctly 1`] = `
<div
  onClick={[Function]}
  to="/"
>
  Link - 
</div>
`;

exports[`when the component is mounted renders correctly with confirmLeave 1`] = `
<div
  onClick={[Function]}
  to="/"
>
  Link - 
</div>
`;
